### PR TITLE
Development

### PR DESCRIPTION
**Pull request by TheCodedProf**
- moved extended tsconfig into our tsconfig to remove errors
- updated some mod checks to allow for running of commands (warn, about) if mod only has manageMessages

Review this ASAP

### CODE REVIEW

**Comment by TheCodedProf** [comment]
Lint & Compile is an issue on discords end. can be ignored

File: src/commands/mod/about.ts [hunk 1]
```diff
@@ -444,7 +444,7 @@ const callback = async (interaction: CommandInteraction): Promise<unknown> => {
 
 const check = (interaction: CommandInteraction) => {
     const member = interaction.member as GuildMember;
-    if (!member.permissions.has("ModerateMembers")) return "You do not have the *Moderate Members* permission";
+    if (!member.permissions.has("ManageMessages")) return "You do not have the *Moderate Members* permission";
```
Comment (Minion3665) [inline src/commands/mod/about.ts:447]: Moderate members must be changed to manage messages in the error text

File: src/commands/mod/warn.ts [hunk 2]
```diff
@@ -300,7 +300,7 @@ const callback = async (
 const check = (interaction: CommandInteraction | ButtonInteraction, partial: boolean = false, target?: GuildMember) => {
     if (!interaction.guild) return;
     const member = interaction.member as GuildMember;
-    if (!member.permissions.has("ModerateMembers")) return "You do not have the *Moderate Members* permission";
+    if (!member.permissions.has("ManageMessages")) return "You do not have the *Moderate Members* permission";
```
Comment (Minion3665) [inline src/commands/mod/warn.ts:303]: And here

File: src/commands/settings/automod.ts [hunk 3]
```diff
@@ -1087,34 +1087,34 @@ const callback = async (interaction: CommandInteraction): Promise<void> => {
         if (i.isButton()) {
             await client.database.guilds.write(interaction.guild.id, { filters: current });
             await client.memory.forceUpdate(interaction.guild.id);
-            config = current;
+            config = _.cloneDeep(current);
             current = _.cloneDeep(config);
```
Comment (Minion3665) [inline src/commands/settings/automod.ts:1091]: Why is an additional deep clone needed here? Given that current is deep cloned before being mutated again it'd seem to me that unless we save a copy of it somewhere there's no need for this

As far as I can tell either this or L1091 is possibly not needed

**Review by Minion3665 — CHANGES_REQUESTED** [review]
Comments in automod do not necessarily need to be heeded (except for efficiency). Other comments do need to be

**Comment by Minion3665** [comment]
> Lint & Compile is an issue on discords end. can be ignored

Are you certain? Discord's end or not it gives an indication of whether we will be able to compile. How do you suggest mitigation of this issue on the server and can we do the same in our test?

**Review by Minion3665 — APPROVED** [review]
Thanks for the fixes. Despite the failed checks we're going to merge, conditional on an investigation ASAP by me

**Pull request merged**